src/libs/Tokenizer.js

```diff
@@ -0,0 +1,95 @@
+
+
+var theTokenizer;
+
+class Tokenizer {
+
+    constructor(program, literalsList) {
+        this.program = program;
+        this.literals = literalsList;
+        this.tokens = [];
+        this.currentToken = 0;
+        this.tokenize();
+    }
+
+    tokenize() {
+        let tokenizedProgram = this.program;
+        tokenizedProgram = tokenizedProgram.replace(/\n/g, "");
+
+        // extracts replacing them with _$_, stores them in an array
+        // to be added back in later.
+        var texts = [...tokenizedProgram.matchAll(/".*"/g)];
+        tokenizedProgram = tokenizedProgram.replace(/".*"/g, "_$_");
+
+        // prevents leading 'lat' in lat lon from combining with variable name token
+        tokenizedProgram = tokenizedProgram.replace(/( \d+)/g, function(m){
+            return "_" + m;
+        });
+        console.log(tokenizedProgram);
+
+        this.literals.forEach((s) => {
+            let regex = new RegExp(s, "g");
+            tokenizedProgram = tokenizedProgram.replace(regex, "_"+s+"_");
+            console.log(tokenizedProgram);
+        });
+        //TODO: Below will make lat and lon hard to distinguish?? may need comma between them??
+        tokenizedProgram = tokenizedProgram.replace(/ /g, "");
```

Comment:
Before this step, we can do `tokenizedProgram.replace(/[-]*\d+\.\d+/g, function(m){
			return "_" + m + "_";
        });`

This encloses each numerical value in the underscore characters. You can try it in the console :)

Reply:
Btw, if we do what I just mentioned above, we won't need to add a comma between the LAT LON values. I'm not saying we shouldn't add the comma in the grammar, because we first have to analyze data from our users, but just pointing this out.

Reply:
hmm... I'll give this a shot later. It seems like what you've put there is probably better than what I wrote.

Reply:
Resolving since you made the change. Thanks!